### PR TITLE
Fixed some typos and gramatical mistakes.

### PR DESCRIPTION
**Pull request by RatanShreshtha**

### CODE REVIEW

File: docs/source/tutorial.rst [hunk 1]
```diff
@@ -760,11 +760,11 @@ restricts parallelism. Of course it would be nice if Python had better
 options for taking advantage of multiple cores, but that's an
 extremely difficult problem to solve, and in the meantime there are
 lots of problems where a single core is totally adequate – or where if
-it isn't, then process- or machine-level parallelism works fine.
+it isn't, then process or machine-level parallelism works fine.
```
Comment (pquentin) [inline docs/source/tutorial.rst]: The intent here is "process-level or machine-level parallelism", so removing the dash changes the meaning here.

**Comment by pquentin** [comment]
Thanks! I don't see any mistakes now, but I'm not a native speaker so I'm not going to merge this myself, sorry. :) Also, maybe @njsmith wants to keep some of the original wording.

**Comment by RatanShreshtha** [comment]
Hi @njsmith please review the PR.

**Comment by njsmith** [comment]
Hey @RatanShreshtha, sorry for missing this before! There was one place where I couldn't decide whether the comma made it better or worse, and eventually decided that the problem was both ways were bad, so I just rewrote it :-). Otherwise looks good to me! Feel free to hit merge once the tests finish (assuming you agree with my changes ;-)).

**Pull request merged**